### ISSUE TITLE
gatk-cnv failed in a T/N project - zero median coverage

### ISSUE THREAD

**Issue by naumenko-sa** [bcbio/bcbio-nextgen#3227]
Hello everyone!

**Version info**
 - bcbio version (bcbio_nextgen.py --version): 1.2.3
 - OS name and version: Ubuntu 18.04.4 LTS / 5.3.0-1017-aws

**To Reproduce**
 - Your [sample configuration file](https://github.com/bcbio/bcbio-nextgen/blob/master/config/examples/somatic_tumor_normal_gatkcnv.yaml)

When testing somatic T/N with gatkcnv one of many samples failed with an [error](
https://gatk.broadinstitute.org/hc/en-us/community/posts/360057785591-Error-while-running-CreateReadCountPanelOfNormals?page=1#community_comment_360010566472).

Tracking here as well if we adjust gatk-cnv wrapper in the result.

SN

**Comment by naumenko-sa** [comment]
A summary from the discussion on the GATK forum and the updated tutorial:
https://gatk.broadinstitute.org/hc/en-us/articles/360035531092--How-to-part-I-Sensitively-detect-copy-ratio-alterations-and-allelic-segments#1

- we should not call CNVs in a T/N pair using the N sample as a PON,
instead, we should create and reuse PONs
- we should check BED files since CollectReadCounts shows <40% non-zero intervals in many samples

I created a PON with 10T/10N samples following the GATK procedure and bcbio.

in gatk-cnv:
- only 3 T/N pairs had wide interval coverage, in 7 samples it was < 40% - bad sequencing; 
`gatk CreateReadCountPanelOfNormals` selected these 3 samples to create a PON, 
and removed the rest; 
- then processing all T samples against PON did not run into
errors, but produced high-quality results only in 3 T samples with wide target coverage, as expected;
- PON is re-used in the rest of the CNV calling steps, including de-noising T samples

in bcbio
-  we are not using PONs to denoise when creating PON itself,
following: https://bcbio-nextgen.readthedocs.io/en/latest/contents/somatic_variants.html#workflow3-somatic-tumor-only-cnvs
https://github.com/bcbio/bcbio-nextgen/blob/58fac9a29797fbff89d095329e4c3a4c8c9b6f89/bcbio/structural/gatkcnv.py#L118
-  denoising samples with low target coverage and without PON leads to a crash:
```
java.lang.IllegalArgumentException: Sample does not have a non-negative sample median.
```
- PON is not created when one sample crashed on denoising step;
- we should not denoise when creating PON; 
- PON is just a merged hdf5 file generated from counts with low-quality samples removed;
- the altered PON procedure gives a quick QC on samples;

**Comment by naumenko-sa** [comment]
upd: creating a gatk-cnv PON with bcbio resulted in an identical PON compared to gatk-cnv outside of bcbio.

Some potential improvements:
- [ ] read counts are calculated sequentially, could be done faster in parallel when having many small samples;
- [ ] to create a PON, we have to use one T sample as a test in addition to normal. Not very intuitive, but maybe it is a feature rather than a bug - related to seq2c use?
- [ ] pon file is not copied to the `final` dir, a user has to dig it out of `work`.

**Comment by naumenko-sa** [comment]
upd
- one T sample for a PON is necessary for cnvkit, that is why it is required in bcbio, it is not needed for gatk-cnv and seq2c
- with PON, gatk-cnv requires T/N pairs to call, T only seems to be not an option

**Comment by naumenko-sa** [comment]
there is a bug: when running PON creation with gatk-cnv + seq2c, or cnvkit + seq2c, the one T sample required gets included in the PON for seq2c
